From e79de413d4b3a81215762e5cf07bb424e578c314 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:16 -0400
Subject: [infiniband] ocrdma: Report actual value of max_fast_reg_page_list_len

Message-id: <350cdb1ecadfc4e443bb8d056585543cdafbc189.1411571481.git.dledford@redhat.com>
Patchwork-id: 96629
O-Subject: [Patch RHEL7 41/48] RDMA/ocrdma: Report actual value of max_fast_reg_page_list_len
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

ocrdma_query_device does not report correct value of max_fast_reg_page_list_len.
This patch applies changes to fix this bug.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Mitesh Ahuja <mitesh.ahuja@emulex.com>
(cherry picked from commit d6a488f21c0f3c44bfbb2339a75159ee55aa2b6f)
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index acb434d..4527311 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -101,7 +101,7 @@ int ocrdma_query_device(struct ib_device *ibdev, struct ib_device_attr *attr)
  attr->max_srq_sge = dev->attr.max_srq_sge;
  attr->max_srq_wr = dev->attr.max_rqe;
  attr->local_ca_ack_delay = dev->attr.local_ca_ack_delay;
- attr->max_fast_reg_page_list_len = 0;
+ attr->max_fast_reg_page_list_len = dev->attr.max_pages_per_frmr;
  attr->max_pkeys = 1;
  return 0;
 }
-- 
1.7.1